core/domain/learner_progress_services.py

```diff
@@ -0,0 +1,415 @@
+# coding: utf-8
+#
+# Copyright 2014 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under tactivities
+"""Services for tracking the progress of the learner."""
+
+from core.domain import subscription_services
+from core.domain import user_domain
+from core.platform import models
+
+(user_models,) = models.Registry.import_models([models.NAMES.user])
+
+
+def get_activities_completed_from_model(activities_completed_model):
+    """Returns an activities completed domain object given a
+    activities completed model loaded from the datastore.
+
+    Args:
+        activities_completed_model: ActivitiesCompletedByLearnerModel. The
+            activities completed model loaded from the datastore.
+
+    Returns:
+        ActivitiesCompletedByLearner: ActivitiesCompletedByLearner. A domain
+        object corresponding to the given model.
+    """
+    return user_domain.ActivitiesCompletedByLearner(
+        activities_completed_model.id,
+        activities_completed_model.completed_exploration_ids,
+        activities_completed_model.completed_collection_ids)
+
+
+def get_incomplete_activities_from_model(incomplete_activities_model):
```

Comment:
Why is this called get_incomplete_activities_... and the previous function called get_activities_completed_...?

There also seems to be a lack of parity in the names of the models / domain objects.